tests/test_edgeql_select.py

```diff
@@ -6274,3 +6286,105 @@ async def test_edgeql_select_reverse_overload_03(self):
                     SELECT User.<whatever
                 ''',
             )
+
+    async def test_edgeql_function_source_01a(self):
+        # TODO: I think we might want to eliminate this sort of shape
```

Comment:
Yes, except, I think it would be the array literal that should erase types.

Reply:
That would be a bit more of a breaking change, since right now shapes will appear through array literals. That might be fine, though?

Reply:
And array_agg also would need the treatment, I think

Reply:
Oh, right now we do some pretty silly stuff: `SELECT [Card {name}, Card {element}]` in json mode will produce output like `[{"name": "Imp"}, {"element": "Fire"}],`

In native mode is just breaks